lib/puppet/provider/sensu_event/sensuctl.rb

```diff
@@ -0,0 +1,95 @@
+require File.expand_path(File.join(File.dirname(__FILE__), '..', 'sensuctl'))
+
+Puppet::Type.type(:sensu_event).provide(:sensuctl, :parent => Puppet::Provider::Sensuctl) do
+  desc "Provider sensu_event using sensuctl"
+
+  mk_resource_methods
+
+  def self.instances
+    events = []
+
+    output = sensuctl_list('event')
+    Puppet.debug("sensu event: #{output}")
+    begin
+      data = JSON.parse(output)
+    rescue JSON::ParserError => e
+      Puppet.debug('Unable to parse output from sensuctl event list')
+      data = []
+    end
+
+    data.each do |d|
+      event = {}
+      event[:ensure] = :present
+      event[:entity] = d['entity']['id']
+      event[:check] = d['check']['name']
+      event[:name] = "#{event[:check]} for #{event[:entity]}"
+      if d['check']['status'] == 0
+        event[:ensure] = :resolve
+      end
+      events << new(event)
+    end
+    events
+  end
+
+  def self.prefetch(resources)
+    events = instances
+    resources.keys.each do |name|
+      if provider = events.find { |e|
+          e.check == resources[name][:check] &&
+          e.entity == resources[name][:entity]
+          }
+        resources[name].provider = provider
+      end
+    end
+  end
+
+  def exists?
+    @property_hash[:ensure] == :present || @property_hash[:ensure] == :resolve
+  end
+
+  def state
+    return @property_hash[:ensure]
+=begin
```

Comment:
what's this begin/end bit about?

Reply:
These are block comments I forgot to remove after testing.